### PR TITLE
Add Neon to database guides

### PR DESCRIPTION
**Pull request by raoufchebri**
## Describe this PR

This guide explains how to connect Prisma to Neon, establish connections when using Prisma Client in serverless functions, and resolve [connection timeout](#connection-timeouts) issues.

### CODE REVIEW

File: content/300-guides/050-database/890-Neon [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+   ```text
+   DATABASE_URL="postgres://daniel:<password>@ep-mute-rain-952417.us-east-2.aws.neon.tech/neondb"
+   ```
+
+   </CodeBlock>
+
+<Admonition type="important">
+If you are using Prisma Client from a serverless function, see [Connect from serverless functions](#connect-from-serverless-functions). To adjust your connection string to avoid connection timeouts issues, see [Connection timeouts](#connection-timeouts).
+</Admonition>
```
Comment (ruheni) [inline content/300-guides/050-database/890-Neon:57]: We could do away with this section.

A section on "Commonalities with other database providers" would be valuable (Example: [Using Prisma with Supabase](https://www.prisma.io/docs/guides/database/supabase#commonalities-with-other-database-providers))

File: content/300-guides/050-database/890-Neon [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+## Connect from serverless functions
```
Comment (ruheni) [inline content/300-guides/050-database/890-Neon:59]: In this section, we could also borrow from the structure used in the Specific considerations section in the "Using Prisma with Supabase" guide. 

We would still distinguish between the two connection strings (pooled and non-pooled) but recommend using the `directUrl` property in the `datasource` block.

This is similar to how it's represented in the Neon docs here: https://neon.tech/docs/guides/prisma-migrate#prisma-migrate-with-pgbouncer

**Review by ruheni — CHANGES_REQUESTED** [review]
I've left two comments. 

TL;DR, let's borrow from the  [Prisma with Supabase guide](https://www.prisma.io/docs/guides/database/supabase#commonalities-with-other-database-providers) and add a section comparing Neon with PostgreSQL, and a "Special considerations" section for information on pooled vs non-pooled connection strings and how to use them with Prisma.

**Comment by ruheni** [comment]
Closing this in favor of #5397 (Continuation of the existing work in the linked PR)
